core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -2580,6 +2583,13 @@ public MasterInodeTtlCheckExecutor() {}
 
     @Override
     public void heartbeat() {
+      try {
+        if (SecurityUtils.isSecurityEnabled() && AuthenticatedClientUser.get() == null) {
+          AuthenticatedClientUser.set(LoginUser.get().getName());
+        }
+      } catch (IOException e) {
+        LOG.error("Failed to set AuthenticatedClientUser in TtlCheckerExecutor heartbeat.");
```

Comment:
It actually should be in `HeartbeatThread.run()` instead. Updated, PTAL. This also fixes `lineageCompleteAndAsyncPersist`.